pkgs/build-support/haxe/override_haxe_lib_from_hmm.nix

```diff
@@ -0,0 +1,103 @@
+# This tool can be used to convert as set of derivation containing haxe dependncies (as from haxe-packages.nix) to use version specified in an hmm.json file, used by certain project to lock inputs.
+# To update (or initiate it) for a package:
+# 1. copy the hmm.json file in the package folder
+# 2. run update_hmm_hashes.py (you will to have nix-prefetch-git in the PATH, which is packaged in nixpkgs under the same name).
+#    It take as argument a path to the hmm.json file, and a path to an output json file that contains the hashes for the dependancies (usually named hashes.json)
+{
+  lib,
+  haxePackages,
+  fetchgit,
+  fetchzip,
+}:
+
+{
+  hmm_attrset ? null,
+  hmm_file ? null,
+  hashes_attrset ? null,
+  hashes_file ? null,
+  packages,
+}:
+
+assert lib.xor (hmm_attrset != null) (hmm_file != null);
+assert lib.xor (hashes_attrset != null) (hashes_file != null);
```

Comment:
I think you can get better error messages if you drop this `xor` assertion -- suggestion below.

But even better would be a function that requires an attrset, and another function that takes in a file and calls `lib.importJSON` from that file.

Having a function with two distinct modes is both more complicated and (I think) unneeded.

Reply:
Indeed. I’ve moved thing to only have a single argument each, and let the caller use the `importJSON` themselve (I ignored the existance of such a fonction before, as a double call seemed too much for a common pattern. Maybe a nixpkgs-wide replacement of `builtins.fromJSON (builtins.readFile(x))`` would be appropriate)

Reply:
Thanks! I'll continue to review.